File: examples/spring-boot-maven-plugin-compile/src/main/java/community/flock/wirespec/example/gradle/app/exception/AppException.java
package community.flock.wirespec.example.gradle.app.exception;

public sealed class AppException extends RuntimeException permits Conflict, NotFound, CallInterrupted, SerializationException {
    public AppException(String message) {
        super(message);
    }

    public AppException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: examples/gradle-ktor/src/main/kotlin/community/flock/wirespec/example/gradle/app/exception/AppException.kt
package community.flock.wirespec.example.gradle.app.exception

import community.flock.wirespec.example.gradle.app.todo.Todo

sealed class AppException(override val message: String) : RuntimeException(message)

sealed class TodoException(message: String) : AppException(message)

class TodoIdNotValidException(invalidId: String) : TodoException("TodoId: $invalidId, not valid")

class TodoNotFoundException(id: Todo.Id) : TodoException("Todo with id: ${id.value}, not found")
